rcl/test/rcl/test_security_directory.cpp

```diff
@@ -109,8 +107,7 @@ TEST_F(TestGetSecureRoot, successScenarios) {
       TEST_SECURITY_DIRECTORY_RESOURCES_DIR_NAME, allocator);
   std::string putenv_input = ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME "=";
   putenv_input += base_lookup_dir_fqn;
-  memcpy(g_envstring, putenv_input.c_str(), sizeof(g_envstring) - 1);
-  putenv_wrapper(g_envstring);
+  putenv_wrapper(putenv_input.c_str());
```

Comment:
I think this pointer has to live as long as the environment variable, and can't be a pointer to a temporary object (like `putenv_input`) on linux.

http://man7.org/linux/man-pages/man3/putenv.3.html

```
In particular, this string
       becomes part of the environment; changing it later will change the
       environment.  (Thus, it is an error to call putenv() with an
       automatic variable as the argument, then return from the calling
       function while string is still part of the environment.)
```

Reply:
@xabxx This one didn't leak (static array). It's probably better to fix the code than changing the behavior this way.

Reply:
changed back the the same behavior and fixed the error.